### DISCUSSION TITLE
Relationship and existing records

### DISCUSSION THREAD

**Discussion by KristofferBroberg** [Budibase/budibase#2761] [Help] [answered]
Hi!

This is probably more of a general database question, but I'm posting it here hoping someone to point me in the right direction.
i
I've got a working relationship in Budibase between two SQL tables. Is there any way I can make Budibase/DB to react on this relationship in regards to the records that existed prior to pulling the db into Budibase? In other words, when I create a new row, the relationship is marked with the "tag" but on the 2000+ earlier records the relationship column is empty.

**Comment by mjashanks** [comment, marked as answer]
Hmm - there shouln't be anything else that you have to do. Apologies if this is a silly question, but are your foreign key columns actually populated properly in your SQL table?

e.g. if you run a `Select * from table1 inner join table2 on table1.foreignKey = table2.id` in your database client (e.g. PGAdmin / MySQL Workbench) - does that work? 

Happy to help you with that `select` statement if you tell me your table names and columns.

**Reply by KristofferBroberg** [reply]
My foreign key column is populated from what I can see...maybe it's not configured correctly.  I'll run the query above and see if it works.

One thing confuses me. I´ve got one in my table...but then, when a relationship is established in Budibase two new columns are created. Is it one of those that you are refering to as the foreign key column?

**Reply by mjashanks** [reply]
yea, it doesn't create a new column in your database, its just an alias. So, to represent as SQL:

```sql
SELECT 
    tasks.* , 
    project.Name as ProjectName 
FROM tasks
INNER JOIN projects on task.projectId = projects.id
```

The `ProjectName` alias above is the "column" that you are referring to.

**Reply by KristofferBroberg** [reply]
OK. Thanks for clarifying!

I ran your select statement in my DB client and it's working as expected.

Here´s the relationship in Budibase

<img width="439" alt="Skärmavbild 2021-09-28 kl  21 55 33" src="https://user-images.githubusercontent.com/5649735/135157093-453ce4db-5243-4211-b47f-43981976571e.png">

**Reply by mjashanks** [reply]
Ok great. So, it's not displaying the relationship properly in budibase? 
1. Can you screenshot the display issue please. 
2. Have you set your table display names? 
3. What DB are you using?

**Reply by KristofferBroberg** [reply]
I just removed the relationship, fetched the tables once again, recreated the relationship and now everything works as expected!

I've done that a couple of times before without any luck. This time around I also renamed the FK column because I realized that it had the exact same name as the PK in the other table. Could that have been the problem?

**Reply by mjashanks** [reply]
Ah - that could be the issue, I will try to reproduce that, and will log as an issue. Thanks!
